pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5296,17 +5296,28 @@ int BPF_KPROBE(trace_security_bpf)
     return events_perf_submit(&data, SECURITY_BPF, 0);
 }
 
-SEC("kprobe/arm_kprobe")
-int BPF_KPROBE(trace_arm_kprobe)
+// arm_kprobe can't be hooked in arm64 architecture, use enable logic instead
+
+static __always_inline int arm_kprobe_handler(struct pt_regs *ctx)
 {
+    args_t saved_args;
+    if (load_args(&saved_args, KPROBE_ATTACH) != 0) {
+        return 0;
+    }
+    del_args(KPROBE_ATTACH);
+
     event_data_t data = {};
     if (!init_event_data(&data, ctx))
         return 0;
 
     if (!should_trace(&data))
         return 0;
 
-    struct kprobe *kp = (struct kprobe *) PT_REGS_PARM1(ctx);
+    struct kprobe *kp = (struct kprobe *) saved_args.args[0];
+    unsigned int retcode = PT_REGS_RC(ctx);
+
+    if (retcode)
+        return 0; // register_kprobe() failed
```

Comment:
@yanivagman should we only submit the event if this is successful OR would you like to have the ret code as an argument to that event ? (@roikol as well).